AppKit/CPWindow/CPWindow.j

```diff
@@ -3844,6 +3814,263 @@ var interpolate = function(fromValue, toValue, progress)
+            var frontmostTrackingArea = overlappingTrackingAreas[0],
+                frontmostView         = [frontmostTrackingArea view];
+            
+            for (var i = 1; i < nbOverlappingTrackingAreas; i++)
+            {
+                var aTrackingArea  = overlappingTrackingAreas[i],
+                    aView          = [aTrackingArea view];
+                
+                // First verify if aView is a subview of frontmostView
+                // If so, it's our new frontmost view
+                
+                var searchingView = aView;
+                
+                while ((searchingView !== _contentView) && ([searchingView superview] !== frontmostView))
+                    searchingView = [searchingView superview];
+                
+                if (searchingView !== _contentView)
+                {
+                    frontmostTrackingArea = aTrackingArea;
+                    frontmostView         = aView;
+                    
+                    continue;
+                }
+                
+                // aView is not a subview of frontmostView
+                // Search in view hierarchy which one will be over the other
+                // (this is done by comparing their draw order)
+                
+                var firstView      = frontmostView,
+                    firstSuperview = [firstView superview];
+                
+                while (firstView !== _contentView)
+                {
+                    var secondView      = aView,
+                        secondSuperview = [secondView superview];
+                    
+                    while ((secondSuperview !== _contentView) && (firstSuperview !== secondSuperview))
+                    {
+                        secondView      = secondSuperview;
+                        secondSuperview = [secondView superview];
+                    }
+                    
+                    if (firstSuperview === secondSuperview)
+                        break;
+                    
+                    firstView      = firstSuperview;
+                    firstSuperview = [firstView superview];
+                }
+                
+                if (firstSuperview !== secondSuperview)
+                    [CPException raise:CPInternalInconsistencyException reason:"Problem with view hierarchy"];
+                
+                var firstViewIndex  = [[firstSuperview subviews] indexOfObject:firstView],
+                    secondViewIndex = [[firstSuperview subviews] indexOfObject:secondView];
+                
+                if (secondViewIndex > firstViewIndex)
+                {
+                    frontmostTrackingArea = aTrackingArea;
+                    frontmostView         = aView;
+                }
+            }
+            
+            if (frontmostTrackingArea !== _activeCursorTrackingArea)
+            {
+                [[frontmostTrackingArea owner] cursorUpdate:[CPEvent enterExitEventWithType:CPCursorUpdate
```

Comment:
You have to check to see if the owner implements cursorUpdate:, since it isn't necessarily a CPResponder subclass.